src/Stratis.Bitcoin.Features.Consensus/Behaviors/ProvenHeadersConsensusManagerBehavior.cs

```diff
@@ -185,6 +185,15 @@ protected override Payload ConstructHeadersPayload(GetHeadersPayload getHeadersP
                         this.logger.LogTrace("(-)[NO_PH_AVAILABLE]");
                         break;
                     }
+                    else if (provenBlockHeader.GetHash() != header.Header.GetHash())
```

Comment:
Hmm I was expecting to compare to prev
```
if (lastHeader != null && lastHeader.Previous.HashBlock != provenBlockHeader.GetHash())
{
    // Proven header in store is not consecutive to proven header in consensus 
    // This can happen in case in case the node is in the middle of a rewind 
    this.logger.LogDebug("PH in store different from consensus '{0}'.", header);
    this.logger.LogTrace("(-)[DIFFERENT_PH_FROM_CONSENSUS]");
    break;
}
```

Reply:
Well come to think about it comparing the hash of current is better I agree